pkg/resources/materialized_view.go

```diff
@@ -0,0 +1,299 @@
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the identifier for the view; must be unique for the schema in which the view is created.",
+	},
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database in which to create the view. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "PUBLIC",
+		Description: "The schema in which to create the view. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"or_replace": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Overwrites the View if it exists.",
+	},
+	"is_secure": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Specifies that the view is secure.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the view.",
+	},
+	"cluster_by": {
+		Type:        schema.TypeList,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Specifies cluster keys for the view with comma separated value format (e.g. \"column1,column2,column3\")",
+	},
+	"statement": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      "Specifies the query used to create the view.",
+		ForceNew:         true,
+		DiffSuppressFunc: DiffSuppressStatement,
+	},
+}
+
+// View returns a pointer to the resource representing a view
+func MaterializedView() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateMaterializedView,
+		Read:   ReadMaterializedView,
+		Update: UpdateMaterializedView,
+		Delete: DeleteMaterializedView,
+
+		Schema: materializedViewSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreateMaterializedView implements schema.CreateFunc
+func CreateMaterializedView(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := d.Get("name").(string)
+	schema := d.Get("schema").(string)
+	database := d.Get("database").(string)
+	s := d.Get("statement").(string)
+
+	builder := snowflake.MaterializedView(name).WithDB(database).WithSchema(schema).WithStatement(s)
+
+	// Set optionals
+	if v, ok := d.GetOk("or_replace"); ok && v.(bool) {
+		builder.WithReplace()
+	}
+
+	if v, ok := d.GetOk("is_secure"); ok && v.(bool) {
+		builder.WithSecure()
+	}
+
+	if v, ok := d.GetOk("comment"); ok {
+		builder.WithComment(v.(string))
+	}
+
+	if v, ok := d.GetOk("cluster_by"); ok {
+		log.Print("[DEBUG] cluster_by ", v)
+		builder.WithClusterBy(v.([]interface{}))
+	}
+
+	q := builder.Create()
+	log.Print("[DEBUG] xxx ", q)
+	err := snowflake.Exec(db, q)
+	if err != nil {
+		return errors.Wrapf(err, "error creating view %v", name)
+	}
+
+	d.SetId(fmt.Sprintf("%v|%v|%v", database, schema, name))
```

Comment:
Rather than prohibit the '|' character, please do what we do for other resources and use the csv library to encode this id. It will properly escape the pipe character.

Reply:
@ryanking Which code use the csv library to encode id?